app/controllers/autocompletes_controller.rb

```diff
@@ -22,11 +47,6 @@ def unique_values_for(field_name, query)
       values << value if value&.match?(matcher)
     end
 
-    check_access(@institution.samples, READ_SAMPLE).select(:id, :core_fields, :custom_fields).find_each do |sample|
-      value = sample.send(field_name)
-      values << value if value&.match?(matcher)
-    end
```

Comment:
Respect to this comment: the only thing that I'm doing here is removing the query on samples and just keeping the query on batches because now (after "reverting" the copy of attributes in samples in #1863) all the autocomplete values belong only to batches. Then this security issue was there before. 

Said that, I will restrict the autocomplete to the institution anyway, mostly because of the performance issue. Leaking data about other institution samples is not super sensitive but is also a plus, so let's go with it :)

Reply:
Oh, right. Forget I said anything.